Client/mods/deathmatch/logic/CPlayerMap.cpp

```diff
@@ -98,26 +102,44 @@ CRadarMap::CRadarMap(CClientManager* pManager)
 
     // Default to attached to player
     SetAttachedToLocalPlayer(true);
+
+    SetupMapVariables();
 }
 
-CRadarMap::~CRadarMap()
+CPlayerMap::~CPlayerMap()
 {
     // Delete our images
-    SAFE_RELEASE(m_pRadarImage);
-    SAFE_RELEASE(m_pLocalPlayerBlip);
+    SAFE_RELEASE(m_mapImageTexture);
+    SAFE_RELEASE(m_playerMarkerTexture);
 
-    for (uint i = 0; i < m_MarkerTextureList.size(); i++)
-        SAFE_RELEASE(m_MarkerTextureList[i]);
+    for (uint i = 0; i < m_markerTextureList.size(); i++)
+        SAFE_RELEASE(m_markerTextureList[i]);
 
-    m_MarkerTextureList.clear();
+    m_markerTextureList.clear();
 
     // Don't need to delete the help texts as those are destroyed by the display manager
 }
 
-void CRadarMap::DoPulse()
+void CPlayerMap::SetMapImage(std::size_t imageIndex)
+{
+    std::uint32_t width, height;
+    if (imageIndex < MAP_IMAGE_SIZES.size())
+        width = height = MAP_IMAGE_SIZES[imageIndex];
+    else // Fail safe, ideally client settings system should not allow this case
+        width = height = MAP_IMAGE_SIZES[0];
+
+    SString fileName("MTA\\cgui\\images\\map_%d.png", width);
+
+    SAFE_RELEASE(m_mapImageTexture);
```

Comment:
Use smart pointers

Reply:
> Use smart pointers

For what variables?

Reply:
> Use smart pointers

std::unique_ptr does not work with CTextureItem

`CTextureItem': cannot instantiate abstract class`

Reply:
What did you try?

Reply:
I showed @TracerDS in DMs, making m_mapImageTexture a std::unique_ptr then using make_unique to create the texture. Wouldn't compile with that error

Reply:
You need `reset` to use raw pointer from `CreateTexture`

Reply:
Its not about that. It causes compilation errors due to the class being too abstract/some argument cannot be converted

Reply:
You cannot directly create an instance of the abstract class `CTextureItem `using `std::unique_ptr` because it’s an abstract class. If there’s a concrete subclass of `CTextureItem`, try creating an instance of that subclass using `std::make_unique`

Reply:
> Its not about that. It causes compilation errors due to the class being too abstract/some argument cannot be converted

It works with `reset`

```c++
    auto* newImage =
        g_pCore->GetGraphics()->GetRenderItemManager()->CreateTexture(...);
    if (newImage)
        m_pRadarImage.reset(newImage);
    else {
    // CreateTexture can return nullptr
    // Handle error
    }
```

Reply:
Will not work.
Compilation error that i showed already `'CTextureItem': cannot instantiate abstract class`